# Importing the libraries

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px

In [2]:
df = pd.read_csv("proteinGroups.csv", low_memory=False)
df.head(10)

Protein IDs Majority protein IDs Peptide counts (all)  \
0  A0A024RBG1           A0A024RBG1                    7   
1  A0A096LP55               P07919           A0A096LP55   
2      P0DPI2           A0A0B4J2D5               P0DPI2   
3      P0DPI3           A0A0U1RR11           A0A0U1RRI6   
4      A0AV96               Q8TBY0               A0AV96   
5      A0AVF1               A0AVF1                    5   
6      A0AVT1               A0AVT1                   35   
7      A0FGR8               A0FGR8                   24   
8      A0JLT2               A0JLT2                    3   
9      A0JNW5               A0JNW5                    3   

  Peptide counts (razor+unique) Peptide counts (unique)  \
0                             2                       2   
1                        P07919                       1   
2                    A0A0B4J2D5                       7   
3                        P0DPI3              A0A0U1RR11   
4                            13                       1   
5                             5                       5   
6                            35                      35   
7                            24                      24   
8                             3                       3   
9                             3                       3   

                                       Protein names Gene names Fasta headers  \
0                                                NaN      NUDT4           NaN   
1                                                  1          1             1   
2                                                  7          7             7   
3                                         A0A0U1RRI6          1             1   
4                                                 13          1            12   
5                Intraflagellar transport protein 56      TTC26           NaN   
6        Ubiquitin-like modifier-activating enzyme 6       UBA6           NaN   
7                           Extended synaptotagmin-2      ESYT2           NaN   
8  Mediator of RNA polymerase II transcription su...      MED19           NaN   
9                       UHRF1-binding protein 1-like  UHRF1BP1L           NaN   

  Number of proteins                Peptides  ... Unnamed: 1014 Unnamed: 1015  \
0                  1                       7  ...           NaN           NaN   
1                  1                       1  ...           NaN           NaN   
2                  7                       7  ...           NaN           NaN   
3                  1                       1  ...           NaN           NaN   
4                  0  RNA-binding protein 47  ...           NaN           NaN   
5                  1                       5  ...           NaN           NaN   
6                  1                      35  ...           NaN           NaN   
7                  1                      24  ...           NaN           NaN   
8                  1                       3  ...           NaN           NaN   
9                  1                       3  ...           NaN           NaN   

  Unnamed: 1016 Unnamed: 1017 Unnamed: 1018 Unnamed: 1019 Unnamed: 1020  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   
5           NaN           NaN           NaN           NaN           NaN   
6           NaN           NaN           NaN           NaN           NaN   
7           NaN           NaN           NaN           NaN           NaN   
8           NaN           NaN           NaN           NaN           NaN   
9           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 1021 Unnamed: 1022 Unnamed: 1023  
0           NaN           NaN           NaN  


# How many proteins were detected in total?

In [3]:
df.shape

(7499, 1024)

In [4]:
df['Protein IDs'].nunique()

7499

In [5]:
df2 = df[~df['Protein IDs'].str.startswith('CON')]
df3 = df2[~df2['Protein IDs'].str.startswith('REV')]
df3['Protein IDs'].nunique()

7251

In [6]:
df3['Protein IDs'].value_counts().sum()

7251

In [7]:
df3.shape

(7251, 1024)

In [8]:
df3.to_csv("my-sample.csv",index_label=False)

In [9]:
df4 = df3['Protein IDs'].dropna(axis=0)

In [10]:
df4.shape

(7251,)

So, total number of unique protein presents are 7251

In [11]:
df3.head()

Protein IDs Majority protein IDs Peptide counts (all)  \
0  A0A024RBG1           A0A024RBG1                    7   
1  A0A096LP55               P07919           A0A096LP55   
2      P0DPI2           A0A0B4J2D5               P0DPI2   
3      P0DPI3           A0A0U1RR11           A0A0U1RRI6   
4      A0AV96               Q8TBY0               A0AV96   

  Peptide counts (razor+unique) Peptide counts (unique) Protein names  \
0                             2                       2           NaN   
1                        P07919                       1             1   
2                    A0A0B4J2D5                       7             7   
3                        P0DPI3              A0A0U1RR11    A0A0U1RRI6   
4                            13                       1            13   

  Gene names Fasta headers Number of proteins                Peptides  ...  \
0      NUDT4           NaN                  1                       7  ...   
1          1             1                  1                       1  ...   
2          7             7                  7                       7  ...   
3          1             1                  1                       1  ...   
4          1            12                  0  RNA-binding protein 47  ...   

  Unnamed: 1014 Unnamed: 1015 Unnamed: 1016 Unnamed: 1017 Unnamed: 1018  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   

  Unnamed: 1019 Unnamed: 1020 Unnamed: 1021 Unnamed: 1022 Unnamed: 1023  
0           NaN           NaN           NaN           NaN           NaN  
1           NaN           NaN           NaN           NaN           NaN  
2           NaN           NaN           NaN           NaN           NaN  
3           NaN           NaN           NaN           NaN           NaN  
4           NaN           NaN           NaN           NaN           NaN  

[5 rows x 1024 columns]

In [12]:
question2 = df3.iloc[:,64:70]

In [13]:
question2

LFQ intensity SW480_1 LFQ intensity SW480_2 LFQ intensity SW480_3  \
0                        0                     0                     0   
1                        0                  28.6                  28.6   
2               7618100000             966260000             812890000   
3                 By MS/MS                   2.9                   2.9   
4                  6139400              29568000             478570000   
...                    ...                   ...                   ...   
7331              27519000              58173000               8690000   
7332              32223000              25792000              26045000   
7333              65504000              56718000              43582000   
7334              22713000              25398000              30747000   
7335             455670000             425850000             466320000   

     LFQ intensity SW620_1 LFQ intensity SW620_2 LFQ intensity Sw620_3  
0                  6943800              16874000              16660000  
1                347180000              57966000              34091000  
2               1904300000            2087000000             735540000  
3                      2.9                   2.9                   2.9  
4                321910000             423610000                     0  
...                    ...                   ...                   ...  
7331              17135000              36965000              29750000  
7332                     0              14617000              20715000  
7333              50852000              43451000              29245000  
7334              68820000              43367000              50953000  
7335             577760000             490020000             513270000  

[7251 rows x 6 columns]

In [14]:
question2['LFQ intensity SW480_1'] = question2['LFQ intensity SW480_1'].str.replace(r"[a-zA-Z].*",'0')
question2['LFQ intensity SW480_2'] = question2['LFQ intensity SW480_2'].str.replace(r"[a-zA-Z].*",'0')
question2['LFQ intensity SW480_3'] = question2['LFQ intensity SW480_3'].str.replace(r"[a-zA-Z].*",'0')
question2['LFQ intensity SW620_1'] = question2['LFQ intensity SW620_1'].str.replace(r"[a-zA-Z].*",'0')
question2['LFQ intensity SW620_2'] = question2['LFQ intensity SW620_2'].str.replace(r"[a-zA-Z].*",'0')
question2['LFQ intensity Sw620_3'] = question2['LFQ intensity Sw620_3'].str.replace(r"[a-zA-Z].*",'0')

In [15]:
question2['LFQ intensity SW480_1']=question2['LFQ intensity SW480_1'].astype('float')
question2['LFQ intensity SW480_2']=question2['LFQ intensity SW480_2'].astype('float')
question2['LFQ intensity SW480_3']=question2['LFQ intensity SW480_3'].astype('float')
question2['LFQ intensity SW620_1']=question2['LFQ intensity SW620_1'].astype('float')
question2['LFQ intensity SW620_2']=question2['LFQ intensity SW620_2'].astype('float')
question2['LFQ intensity Sw620_3']=question2['LFQ intensity Sw620_3'].astype('float')
print(question2['LFQ intensity SW480_1'])
print(question2['LFQ intensity SW480_2'])
print(question2['LFQ intensity SW480_3'])
print(question2['LFQ intensity SW620_1'])
print(question2['LFQ intensity SW620_2'])
print(question2['LFQ intensity Sw620_3'])

0       0.000000e+00
1       0.000000e+00
2       7.618100e+09
3       0.000000e+00
4       6.139400e+06
            ...     
7331    2.751900e+07
7332    3.222300e+07
7333    6.550400e+07
7334    2.271300e+07
7335    4.556700e+08
Name: LFQ intensity SW480_1, Length: 7251, dtype: float64
0               0.0
1              28.6
2       966260000.0
3               2.9
4        29568000.0
           ...     
7331     58173000.0
7332     25792000.0
7333     56718000.0
7334     25398000.0
7335    425850000.0
Name: LFQ intensity SW480_2, Length: 7251, dtype: float64
0               0.0
1              28.6
2       812890000.0
3               2.9
4       478570000.0
           ...     
7331      8690000.0
7332     26045000.0
7333     43582000.0
7334     30747000.0
7335    466320000.0
Name: LFQ intensity SW480_3, Length: 7251, dtype: float64
0       6.943800e+06
1       3.471800e+08
2       1.904300e+09
3       2.900000e+00
4       3.219100e+08
            ...     
7331    1.713500e+07
7332    

In [16]:
question2.head(10)

LFQ intensity SW480_1  LFQ intensity SW480_2  LFQ intensity SW480_3  \
0           0.000000e+00           0.000000e+00           0.000000e+00   
1           0.000000e+00           2.860000e+01           2.860000e+01   
2           7.618100e+09           9.662600e+08           8.128900e+08   
3           0.000000e+00           2.900000e+00           2.900000e+00   
4           6.139400e+06           2.956800e+07           4.785700e+08   
5           2.628000e+07           2.719900e+07           2.667900e+07   
6           1.062700e+09           1.101700e+09           1.349900e+09   
7           7.257900e+08           7.305000e+08           8.185200e+08   
8           6.553100e+07           5.569900e+07           8.127500e+07   
9           1.827300e+07           1.758000e+07           2.340000e+07   

   LFQ intensity SW620_1  LFQ intensity SW620_2  LFQ intensity Sw620_3  
0           6.943800e+06           1.687400e+07           1.666000e+07  
1           3.471800e+08           5.796600e+07           3.409100e+07  
2           1.904300e+09           2.087000e+09           7.355400e+08  
3           2.900000e+00           2.900000e+00           2.900000e+00  
4           3.219100e+08           4.236100e+08           0.000000e+00  
5           3.570900e+07           3.516900e+07           3.359500e+07  
6           1.488500e+09           1.139000e+09           1.086000e+09  
7           8.441200e+08           8.569700e+08           8.187600e+08  
8           9.684400e+07           6.870400e+07           9.606200e+07  
9           1.880500e+07           1.640200e+07           1.703800e+07

In [17]:
question2.shape

(7251, 6)

In [18]:
question2.count()

LFQ intensity SW480_1    7247
LFQ intensity SW480_2    7248
LFQ intensity SW480_3    7245
LFQ intensity SW620_1    7248
LFQ intensity SW620_2    7249
LFQ intensity Sw620_3    7251
dtype: int64

In [22]:
question2['LFQ intensity SW480_1'].replace('', np.nan, inplace=True)
question2['LFQ intensity SW480_2'].replace('', np.nan, inplace=True)
question2['LFQ intensity SW480_3'].replace('', np.nan, inplace=True)
question2['LFQ intensity SW620_1'].replace('', np.nan, inplace=True)
question2['LFQ intensity SW620_2'].replace('', np.nan, inplace=True)
question2['LFQ intensity Sw620_3'].replace('', np.nan, inplace=True)

In [23]:
question2.isnull().sum()

LFQ intensity SW480_1    4
LFQ intensity SW480_2    3
LFQ intensity SW480_3    6
LFQ intensity SW620_1    3
LFQ intensity SW620_2    2
LFQ intensity Sw620_3    0
dtype: int64

In [ ]:
question2_clean = pd.read_csv("")

In [ ]:
question2 = question2.dropna()

In [ ]:
question2

In [24]:
question2.to_csv("question2.csv")

In [ ]:
np.count_nonzero(question2, axis=0)